src/nnfusion/core/operators/generic_op/generic_op_define/Convolution.cpp

```diff
@@ -27,38 +28,50 @@ REGISTER_OP(Convolution)
              {"padding", vector_to_string(padding)},
              {"dilation", vector_to_string(dilation)}});
     })
+    */
     .translate_v2([](std::shared_ptr<graph::GNode> curr) -> std::string {
         auto ir_template =
             R"( @output0@@output0_layout@ +=! @input0@@input0_layout@@pad_cond@ * @input1@@input1_layout@ where HO in @height@, WO in @width@; )";
         auto manual_rule = R"( ## @: plan/convfwd_@data_format@_v1 )";
 
         auto _op = static_pointer_cast<nnfusion::op::Convolution>(curr->get_op_ptr());
         NNFUSION_CHECK_NOT_NULLPTR(_op) << "Node type is not " << curr->get_op_ptr()->get_op_type();
-
+        const auto& dilation_h = _op->get_window_dilation_strides()[0];
+        const auto& dilation_w = _op->get_window_dilation_strides()[1];
+        const auto& stride_h = _op->get_window_movement_strides()[0];
+        const auto& stride_w = _op->get_window_movement_strides()[1];
         const auto& is_nchw = _op->get_data_format() == "NCHW";
         const auto& padding_h = _op->get_padding_below()[0];
         const auto& padding_w = _op->get_padding_below()[1];
+        const auto& kernel_size_h =
+            is_nchw ? curr->get_input_shape(1)[2] : curr->get_input_shape(1)[0];
+        const auto& kernel_size_w =
+            is_nchw ? curr->get_input_shape(1)[3] : curr->get_input_shape(1)[1];
         const auto& in_shape = curr->get_input_shape(0);
         const std::string data_format = is_nchw ? "nchw" : "nhwc";
-
+        NNFUSION_CHECK(dilation_h == 1) << "Not support other dilation yet.";
+        NNFUSION_CHECK(dilation_w == 1) << "Not support other dilation yet.";
         nnfusion::op::OpConfig::any config;
-        auto shape_template = is_nchw ? "[N, C, -@pad_0@ + HO + KH, -@pad_1@ + WO + KW]"
-                                      : "[N, -@pad_0@ + HO + KH, -@pad_1@ + WO + KW, C]";
+        std::string HO = "-@pad_0@ + KH + HO * " + to_string(stride_h);
+        std::string WO = "-@pad_1@ + KW + WO * " + to_string(stride_w);
+        std::string shape_template =
+            is_nchw ? "[N, C, " + HO + ", " + WO + "]" : "[N, " + HO + ", " + WO + ", C]";
         config["input1_layout"] = is_nchw ? "[F, C, KH, KW]" : "[KH, KW, C, F]";
```

Comment:
input_shape1 should be [F, C, KH, KW] in both NCHW and NHWC format because NCHW and NHWC only indicate the shape format of input0 in Convolution op.

Reply:
> input_shape1 should be [F, C, KH, KW] in both NCHW and NHWC format because NCHW and NHWC only indicate the shape format of input0 in Convolution op.

it is true in depthwisecon2d op, but in conv op, input1 shape goes with dataformat.  (refer to https://github.com/microsoft/nnfusion/blob/master/src/nnfusion/core/operators/util/validation_util.cpp line 192:199)

Reply:
I think this is because the validation_util is implemented according to current conv op implementation in NNFusion. The [KH, KW, C, F] filter layout is TensorFlow's filter format. In TensorFlow, both "NCHW" and "NWHC" have [KH, KW, C, F] filter layout. The input layout and the filter layout are decoupled in other frameworks like tvm.relay. The "NCHW" format only indicates the input layout.

Reply:
> I think this is because the validation_util is implemented according to current conv op implementation in NNFusion. The [KH, KW, C, F] filter layout is TensorFlow's filter format. In TensorFlow, both "NCHW" and "NWHC" have [KH, KW, C, F] filter layout. The input layout and the filter layout are decoupled in other frameworks like tvm.relay. The "NCHW" format only indicates the input layout.

@xysmlx Our conv op define inherits from ngraph. I guess it is because the filter shape goes with dataformat in ngraph. As for kernel, our local cuda kernel only accepts NCHW, and it assumes filter shape = [F, C, KH, KW], while the antares kernel accepts both NCHW and NHWC, but in the schedular, it assumes config["input1_layout"] = is_nchw ? "[F, C, KH, KW]" : "[KH, KW, C, F]".   refer to https://github.com/microsoft/antares/blob/master/platforms/c-rocm/schedule/standard/convfwd_nhwc_v1.py line 99 and https://github.com/microsoft/antares/blob/master/platforms/c-rocm/schedule/standard/convfwd_nchw_v1.py line 98. We could decouple the input layout and the filter layout in our conv2d op, but we need to modify both the conv2d op define( especially valid and infer shape()) and the kernels.

Reply:
NGraph's convolution only supports NCHW format and the filter layout is [F, C, KH, KW] (https://github.com/NervanaSystems/ngraph/blob/master/src/ngraph/op/convolution.hpp). But current implementation does not affect this release. I think we could decouple them and support more attributes in the Convolution op (e.g., asymmetric padding and stride) in future releases.

Reply:
> NGraph's convolution only supports NCHW format and the filter layout is [F, C, KH, KW] (https://github.com/NervanaSystems/ngraph/blob/master/src/ngraph/op/convolution.hpp). But current implementation does not affect this release. I think we could decouple them and support more attributes in the Convolution op (e.g., asymmetric padding and stride) in future releases.

agree